### PR TITLE
terraform-providers.teleport: 7.3.0 -> 8.0.6

### PR DESCRIPTION
**Pull request by justinas**
###### Motivation for this change

Corresponding to the new major version of Teleport proper. (#149085).

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ymatsiuk** [comment]
Result of `nixpkgs-review pr 150964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>terraform-providers.teleport</li>
  </ul>
</details>

**Review by ymatsiuk — APPROVED** [review]

**Review by risicle — APPROVED** [review]
`nixpkgs-review` happy, nixos x86_64 & aarch64.

`teleport` itself doesn't build on macos at the moment, so this change prevents this building there. Guess it just means we need to fix `teleport`.

**Comment by justinas** [comment]
> `teleport` itself doesn't build on macos at the moment, so this change prevents this building there. Guess it just means we need to fix `teleport`.

Ah, that sucks. I'll see if I can get a Mac machine to debug the failure.

**Comment by ymatsiuk** [comment]
teleport build fine on mac, the test are failing 🤔 shall we just disable test for mac?

**Comment by risicle** [comment]
I haven't looked deeply at this but it looks like it's simply failing at

```
mkdir /var/empty/.tsh: permission denied
```

My normal fix for this would be setting `$HOME` to something writable, but it looks like we already do that. It's possible it's digging our users "real" home out of `/etc/passwd` as the openssh test suite does, which required a bit of `libredirect` trickery...

**Comment by ymatsiuk** [comment]
Result of `nixpkgs-review pr 150964` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>terraform-providers.teleport</li>
  </ul>
</details>

**Comment by ymatsiuk** [comment]
Result of `nixpkgs-review pr 150964` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>terraform-full</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>terraform-providers.teleport</li>
  </ul>
</details>

**Review by ymatsiuk — APPROVED** [review]

**Comment by justinas** [comment]
Teleport 8.0.6 has been merged and this has been rebased on top of it. Should be good to go.

File: pkgs/applications/networking/cluster/terraform-providers/teleport/default.nix [hunk 1]
```diff
@@ -1,17 +1,19 @@
...
   src = fetchFromGitHub {
     owner = "gravitational";
     repo = "teleport-plugins";
     rev = "v${version}";
-    sha256 = "19zn78nn64gc0nm7ycblzi4549a0asql07pfxvrphi6s9fjr5m3y";
+    sha256 = "1rhvpbw4dga256dp2cr5f912d2j7rh8pd1v88dlgq3mmw8n5c7vy";
   };
   vendorSha256 = null;
 
+  nativeBuildInputs = [ teleport ];
```
Comment (zowoq) [inline pkgs/applications/networking/cluster/terraform-providers/teleport/default.nix:15]: Doesn't seem to be needed during the build.

```suggestion
  checkInputs = [ teleport ];
```
Also the generic `terraform-provider` build now disables `doCheck` by default so you could do that here and drop the `teleport` input entirely.

Comment (justinas) [reply pkgs/applications/networking/cluster/terraform-providers/teleport/default.nix]: Thanks, was not aware of `checkInputs`. I left the check phase enabled for now, might disable if it becomes more problematic in the future.

**Pull request merged**